examples/computer_vision/detr/model_def.py

```diff
@@ -0,0 +1,201 @@
+"""
+"""
+from collections import defaultdict
+from typing import Any, Dict, Sequence, Union
+from attrdict import AttrDict
+import numpy as np
+import sys
+
+sys.path.append("./detr")
+
+import torch
+
+from determined.pytorch import (
+    DataLoader,
+    LRScheduler,
+    PyTorchTrial,
+    PyTorchTrialContext,
+)
+
+# DETR imports
+import detr.util.misc as utils
+from detr.datasets import get_coco_api_from_dataset
+from detr.datasets.coco_eval import CocoEvaluator, create_common_coco_eval
+from model import build_model
+
+# Experiment dir imports
+from data import unwrap_collate_fn, build_dataset
+
+
+TorchData = Union[Dict[str, torch.Tensor], Sequence[torch.Tensor], torch.Tensor]
+
+
+class DETRTrial(PyTorchTrial):
+    def __init__(self, context: PyTorchTrialContext) -> None:
+        self.context = context
+        self.hparams = AttrDict(self.context.get_hparams())
+
+        # Wrap the model.
+        model, self.criterion, self.postprocessors = build_model(
+            self.hparams, world_size=self.context.distributed.get_size()
+        )
+        self.model = self.context.wrap_model(model)
+
+        n_parameters = sum(p.numel() for p in model.parameters() if p.requires_grad)
+        print("number of params:", n_parameters)
+
+        param_dicts = [
+            {
+                "params": [
+                    p
+                    for n, p in self.model.named_parameters()
+                    if "backbone" not in n and p.requires_grad
+                ]
+            },
+            {
+                "params": [
+                    p
+                    for n, p in self.model.named_parameters()
+                    if "backbone" in n and p.requires_grad
+                ],
+                "lr": self.hparams.lr_backbone,
+            },
+        ]
+        self.optimizer = self.context.wrap_optimizer(
+            torch.optim.AdamW(
+                param_dicts, lr=self.hparams.lr, weight_decay=self.hparams.weight_decay
+            )
+        )
+
+        # Wrap the LR scheduler.
+        self.lr_scheduler = self.context.wrap_lr_scheduler(
+            torch.optim.lr_scheduler.StepLR(self.optimizer, self.hparams.lr_drop),
+            step_mode=LRScheduler.StepMode.STEP_EVERY_EPOCH,
+        )
+
+        self.clip_grads_fn = (
```

Comment:
I don't see this function used anywhere else. Can we delete it if that's the case?